### PR TITLE
Update API doc for feature PodTopologySpread (a.k.a EvenPodsSpread)

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind documentation
/sig scheduling

**What this PR does / why we need it**:

This PR removes the "alpha" wordings on feature PodTopologySpread (a.k.a EvenPodsSpread) as it's promoted to beta in 1.18.

**Which issue(s) this PR fixes**:

Part of #87662.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-scheduling/20190221-pod-topology-spread.md
```

### CODE REVIEW

**Comment by Huang-Wei** [comment]
/assign @lavalamp 

Daniel, it's a simple doc update, could you kindly take a look?

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -3038,8 +3038,7 @@ type PodSpec struct {
 	Overhead ResourceList `json:"overhead,omitempty" protobuf:"bytes,32,opt,name=overhead"`
 	// TopologySpreadConstraints describes how a group of pods ought to spread across topology
 	// domains. Scheduler will schedule pods in a way which abides by the constraints.
-	// This field is alpha-level and is only honored by clusters that enables the EvenPodsSpread
-	// feature.
+	// This field is only honored by clusters that enables the EvenPodsSpread feature.
```
Comment (lavalamp) [inline staging/src/k8s.io/api/core/v1/types.go]:
```suggestion
	// This field is only honored by clusters that enable the EvenPodsSpread feature.
```

**Comment by lavalamp** [comment]
Here's an approve, please fix typo & get lgtm from anyone. Thanks!

/approve

**Comment by Huang-Wei** [comment]
@ahg-g Can you /lgtm this?

File: api/openapi-spec/swagger.json [hunk 2]
```diff
@@ -8553,7 +8553,7 @@
           "type": "array"
         },
         "topologySpreadConstraints": {
-          "description": "TopologySpreadConstraints describes how a group of pods ought to spread across topology domains. Scheduler will schedule pods in a way which abides by the constraints. This field is alpha-level and is only honored by clusters that enables the EvenPodsSpread feature. All topologySpreadConstraints are ANDed.",
+          "description": "TopologySpreadConstraints describes how a group of pods ought to spread across topology domains. Scheduler will schedule pods in a way which abides by the constraints. This field is only honored by clusters that enable the EvenPodsSpread feature. All topologySpreadConstraints are ANDed.",
```
Comment (Huang-Wei) [inline api/openapi-spec/swagger.json]: The tricky thing is `hack/update-openapi-spec.sh` only generates this line on a Linux machine (doesn't work on Mac). Now CI should be green.

**Pull request merged**